hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOMRatisSnapshots.java

```diff
@@ -1036,6 +1069,292 @@ public void testInstallCorruptedCheckpointFailure() throws Exception {
+        OM_KEY_PREFIX + ozoneBucket.getName() +
+        OM_KEY_PREFIX + newKeys.get(0);
+    Table<String, OmKeyInfo> omKeyInfoTableBeforeDeletion = newLeaderOM
+        .getMetadataManager()
+        .getKeyTable(ozoneBucket.getBucketLayout());
+    OmKeyInfo newKeyInfo = omKeyInfoTableBeforeDeletion.get(newKey);
+    Assertions.assertNotNull(newKeyInfo);
+
+    long usedBytes = getUsedBytes(newLeaderOM);
+
+    ozoneBucket.deleteKeys(newKeys);
+
+    GenericTestUtils.waitFor(() -> {
+      try {
+        return getUsedBytes(newLeaderOM) ==
+            usedBytes - newKeyInfo.getDataSize();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 1000, 10000);
+
+    GenericTestUtils.waitFor(() -> {
+      Table<String, OmKeyInfo> omKeyInfoTableAfterDeletion = newLeaderOM
+          .getMetadataManager()
+          .getKeyTable(ozoneBucket.getBucketLayout());
+      try {
+        return Objects.isNull(omKeyInfoTableAfterDeletion.get(newKey));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 1000, 10000);
+
+    // Check whether newly created snapshot data is reclaimed
+    client.getObjectStore()
+        .deleteSnapshot(volumeName, bucketName, newSnapshot.getName());
+    GenericTestUtils.waitFor(() -> {
+      Table<String, SnapshotInfo> snapshotInfoTable =
+          newLeaderOM.getMetadataManager().getSnapshotInfoTable();
+      try {
+        return null == snapshotInfoTable.get(newSnapshot.getTableKey());
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }, 1000, 10000);
+
+    // Check whether compaction logs get appeneded to
+    // Force compaction
+    newLeaderOM.getMetadataManager()
+        .getStore()
+        .compactDB();
+    int newNumberOfLogFiles = 0;
+    long newContentLength;
+    try (BufferedReader bufferedReader =
+             Files.newBufferedReader(currentCompactionLog);
+         DirectoryStream<Path> files =
+             Files.newDirectoryStream(compactionLogPath)) {
+      newContentLength = bufferedReader.lines()
+          .mapToLong(String::length)
+          .reduce(0L, Long::sum);
+      for (Path ignored : files) {
+        newNumberOfLogFiles++;
+      }
+    }
+    Assertions.assertTrue(numberOfLogFiles != newNumberOfLogFiles
+        || contentLength != newContentLength);
+
+    // Check whether sst files were pruned
+    final int finalNumberOfSstFiles = numberOfSstFiles;
+    try {
+      GenericTestUtils.waitFor(() -> {
+        int newNumberOfSstFiles = 0;
+        try (DirectoryStream<Path> files =
+                 Files.newDirectoryStream(sstBackupDirPath)) {
+          for (Path ignored : files) {
+            newNumberOfSstFiles++;
+          }
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+        return finalNumberOfSstFiles > newNumberOfSstFiles;
+      }, 1000, 10000);
+    } catch (TimeoutException e) {
```

Comment:
like the one above, this catch clause seems to defeat the purpose of the test.  Is the test flaky without the catch clause?

Reply:
(Please run the 100 iterations in the ci, like we did earlier.)

Reply:
This one is flaky without it. The reason is that it is hard to set interval of pruning correctly because tests before does not execute in constant time.

Reply:
As per [smenglc comment](https://github.com/apache/ozone/pull/5083#discussion_r1270234790) I moved this change to separate dev branch [https://github.com/mladjan-gadzic/ozone/tree/HDDS-8709-4](https://github.com/mladjan-gadzic/ozone/tree/HDDS-8709-4).